exercises/binary-search-tree/canonical-data.json

```diff
@@ -0,0 +1,178 @@
+{
+    "exercise": "binary-search-tree",
+    "version": "1.0.0",
+    "comments": [
+        "Each test case assumes an empty/new tree.",
+        "As per exercism/problem-specifications#996 key 'tree_data' counts as an input",
+        "to test generators.",
+        "The key 'tree_data' represents an array of numbers for which the data should be ",
+        "inserted/added to the tree as it appears in the array from left to right.",
+        "e.g. tree_data: ['2', '1', '3', '6', '7', '5']",
+        "Insert 2. Insert 1. Insert 3. Insert 6, so on...",
+        "This canonical-data does not restrict the data type of array elements to either",
+        "strings or integers.",
+        "The key 'expected' represents tree state as JSON object of schema :",
+        "{",
+        "    'title':'nodeObject',",
+        "    'type':'object',",
+        "    'properties':{",
+        "        'data':{",
+        "            'type':'string'",
+        "        },",
+        "        'left':{",
+        "            'type':'nodeObject'",
+        "        },",
+        "        'right':{",
+        "            'type':'nodeObject'",
+        "        }",
+        "    },",
+        "    'required':['data', 'left', 'right']",
+        "}"
+    ],
+    "cases": [{
+            "description": "data is retained",
+            "property": "data",
+            "input": {
+                "tree_data": ["4"]
+            },
+            "expected": {
+                "data": "4",
+                "left": null,
+                "right": null
+            }
+        },
+        {
+            "description": "insert data at proper node",
+            "cases": [{
+                    "description": "smaller number at left node",
+                    "property": "data",
+                    "input": {
+                        "tree_data": ["4", "2"]
+                    },
+                    "expected": {
+                        "data": "4",
+                        "left": {
+                            "data": "2",
+                            "left": null,
+                            "right": null
+                        },
+                        "right": null
+                    }
+                },
+                {
+                    "description": "same number at left node",
+                    "property": "data",
+                    "node": "root->left",
```

Comment:
I think in this file you have taken your suggestion of https://github.com/exercism/problem-specifications/pull/940#issuecomment-341854357 to use `expected` to represent the entire tree, and not your suggestion of https://github.com/exercism/problem-specifications/pull/940#issuecomment-341592212 to list one specific node being traversed to. If you confirm this is true, I think this `node` line should be removed.

I don't think `node` is still necessary, right? Now that `expected` shows the full tree.

Reply:
Yes!, good catch.